### PR TITLE
frontend: convert headerssync to typescript

### PR DESCRIPTION
**Pull request by thisconnect**

### CODE REVIEW

**Comment by thisconnect** [comment]
PTAL

File: frontends/web/src/components/headerssync/headerssync.tsx [hunk 1]
```diff
@@ -1,5 +1,7 @@
+
 /**
- * Copyright 2018 Shift Devices AG
+ * Copyright 2018  Shift Devices AG
+ * Copyright 2020 Shift Crypto AG
```
Comment (x1ddos) [inline frontends/web/src/components/headerssync/headerssync.tsx:4]: 2021?

File: frontends/web/src/components/headerssync/headerssync.tsx [hunk 2]
```diff
@@ -14,44 +16,61 @@
...
 import * as style from './headerssync.css';
 
-@translate()
-// @ts-ignore (generics need to be typed explicitly once converted to TypeScript)
-@subscribe(props => ({ status: 'coins/' + props.coinCode + '/headers/status' }))
-export default class HeadersSync extends Component {
-    constructor(props) {
-        super(props);
-        this.state = { show: 0 };
+interface HeadersSyncProps {
```
Comment (x1ddos) [inline frontends/web/src/components/headerssync/headerssync.tsx:26]: So, are the interface types prefixed with `I` or not. What's the new convention?

File: frontends/web/src/components/headerssync/headerssync.tsx [hunk 3]
```diff
@@ -14,44 +16,61 @@
...
 
     componentDidUpdate(prevProps) {
-        const status = this.props.status;
+        const { status } = this.props;
         if (status && prevProps.status && status.tip !== prevProps.status.tip) {
-            this.setState({ show: status.tip }); // eslint-disable-line
+            this.setState({ show: status.tip });
             if (status.tip === status.targetHeight) {
-                setTimeout(() => this.setState(state => state.show === status.tip && { show: 0 }), 4000);
+                setTimeout(() => this.setState(state => state.show === status.tip ? { show: 0 } : null), 4000);
```
Comment (x1ddos) [inline frontends/web/src/components/headerssync/headerssync.tsx:57]: Since w'ere here, what's this magic 4000ms? I mean, why 4000?

**Review by x1ddos — APPROVED** [review]
LGTM

File: frontends/web/src/components/headerssync/headerssync.tsx (hunk 1 above)
Comment (thisconnect) [reply frontends/web/src/components/headerssync/headerssync.tsx:4]: well at the time it was right :)

Screen Shot 2021-02-10 at 4 11 44 PM

File: frontends/web/src/components/headerssync/headerssync.tsx (hunk 3 above)
Comment (thisconnect) [reply frontends/web/src/components/headerssync/headerssync.tsx:58]: commented `hide component after 4s when tip reached targetHeight`, but no idea why 4s was chosen and not 5s.

It used to be 5s :) https://github.com/digitalbitbox/bitbox-wallet-app/commit/499ad0f2ed6d956c261e63b2abee9ac6b2fc8456#diff-11692a3eba9e705270b5c20ffbad8be0ca2b85a4e380a0f3a945de78c442859b

File: frontends/web/src/components/headerssync/headerssync.tsx (hunk 2 above)
Comment (thisconnect) [reply frontends/web/src/components/headerssync/headerssync.tsx]: sure, we had it only for the exported interfaces that are going to be shared everywhere, but fine by me if we start using it also for internal stuff.

**Pull request merged**